Samples/PackageWithExternalLocation/cppwinrt/PackageWithExternalLocationCppAppSetup/main.cpp

```diff
@@ -0,0 +1,80 @@
+﻿#include "pch.h"  
+#include <iostream>  
+#include <string>
+#include <winrt/Windows.Management.Deployment.h>  
+#include <winrt/Windows.ApplicationModel.Activation.h>  
+#include <shellapi.h>
+
+using namespace winrt;  
+using namespace winrt::Windows::Management::Deployment;  
+
+void InstallPackage(const std::wstring& packagePath, const std::wstring& externalLocation)
+{
+    try 
+    {
+        if (!externalLocation.empty() && !packagePath.empty())
+        {
+            winrt::Windows::Management::Deployment::PackageManager packageManager;
+            winrt::Windows::Management::Deployment::AddPackageOptions addOptions;
+            addOptions.ExternalLocationUri(winrt::Windows::Foundation::Uri(winrt::hstring(externalLocation)));
+
+            auto deploymentResult = packageManager.AddPackageByUriAsync(winrt::Windows::Foundation::Uri(winrt::hstring(packagePath)), addOptions).get();
+            auto extendedErrorCode = deploymentResult.ExtendedErrorCode();
+
+            if (extendedErrorCode != HRESULT_FROM_WIN32(ERROR_PACKAGES_IN_USE) && FAILED(extendedErrorCode))
+            {
+                if (extendedErrorCode == HRESULT_FROM_WIN32(CERT_E_UNTRUSTEDROOT))
+                {
+                    MessageBox(nullptr, L"Please install your certificate to Trusted Root Certification Authorities for Local Machine. Running without identity", L"result", MB_ICONINFORMATION);
```

Comment:
`L"result"` will literally print the string "result" -- what you need to do is somthing like:

```C++
wchar_t buff[10];
_itow_s(result, buff, 16);
```
and then use `buff` in your call.